hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,434 @@
+
+# Copyright 2013-2017 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied.  See the License for the specific language
+# governing permissions and limitations under the License.
+"""A switch driver for the juniper QFX series.
+
+But in theory can work for any Juniper switch that supports NETCONF.
+Driver uses the python library PyEZ made available by Juniper.
+By default it sets NETCONF session over ssh, i.e. make RPC calls over ssh.
+For this to work, make sure that the Juniper switch is configured
+appropriately to recieve NETCONF connections. Its default port is 830.
+"""
+
+import re
+import logging
+import schema
+import json
+import sys
+import os
+
+from jnpr.junos import Device
+from jnpr.junos.exception import ConnectError
+from jnpr.junos.utils.config import Config
+from jnpr.junos.utils.config import ConfigLoadError
+
+
+from hil.model import db, Switch, BigIntegerType
+from hil.migrations import paths
+from hil import config
+from hil.config import cfg
+from os.path import dirname, join
+
+import pdb; pdb.set_trace()
+config.load()
+j_driver = 'hil.ext.switches.juniper'
+if cfg.has_option('extensions', j_driver):
+    template_dir = cfg.get('extensions', j_driver)
+    sys.path.append(template_dir)
+    try:
+        from config_tables.ConfigTables import InterfaceConfigTable
+    except ImportError:
+        pass
+
+logger = logging.getLogger(__name__)
+paths[__name__] = join(dirname(__file__), 'migrations', 'juniper')
+
+
+class VlanAddError(Exception):
+    """ Raise this exception when vlan cannot be added to the port."""
+    pass
+
+
+class ConfigCommitError(Exception):
+    """ Raise this exception if there are commit conflicts."""
+    pass
+
+
+class Juniper(Switch):
```

Comment:
inherit from both `Switch` and `SwitchSession`. [Example](https://github.com/CCI-MOC/hil/blob/master/hil/ext/switches/dellnos9.py#L38).

Reply:
see this comment [here](https://github.com/CCI-MOC/hil/pull/850#discussion_r144900753)

If that's the case, then the Juniper class will inherit from Switch like you did. And then there will be a class for Session which inherits from SwitchSession.